src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,130 @@
+                    for (int i = 0; i < tasks.getLength(); i++) {
+                        fw.write(tasks.getTask(i).savedString() + System.lineSeparator());
+                    }
+                    fw.close();
+                } catch (IOException e) {
+                    Ui.printCannotOpenFile();
+                }
+            } else {
+                try {
+                    FileWriter fw = new FileWriter(filePath);
+                    for (int i = 0; i < tasks.getLength(); i++) {
+                        fw.write(tasks.getTask(i).savedString() + System.lineSeparator());
+                    }
+                    fw.close();
+                } catch (IOException e) {
+                    Ui.printUnableToAppend();
+                }
+            }
+    }
+
+    public TaskList load() {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+            int index = 1;
+            File f = new File(filePath);
+
+            if (!f.exists()) {
+                return new TaskList(tasks, new Ui());
+            }
+
+
+        Scanner sc = null;
+
+        try {
+            sc = new Scanner(f);
+        } catch (FileNotFoundException e) {
+            Ui.printFileNotFound();
+        }
+
+        while (sc.hasNext()) {
+
+            String input = sc.nextLine();
+
+                if (input.startsWith("T")) {
+                    String desc = descStr(input);
+                    if (String.valueOf(input.charAt(1)) == "Y") {
+                        tasks.add(new ToDo(true, desc, index));
+                    } else {
+                        tasks.add(new ToDo(false, desc, index));
+                    }
+                    index++;
+                    continue;
+                }
+                if (input.startsWith("D")) {
+                    String desc = descStr(input);
+                    String time = timeStr(input);
+                    LocalDate date = dateStr(input);
+                    if (String.valueOf(input.charAt(1)) == "Y") {
+                        tasks.add(new Deadline(true, desc, index, date, time));
+                    } else {
+                        tasks.add(new Deadline(false, desc, index, date, time));
+                    }
+                    index++;
+                    continue;
+                }
+                if (input.startsWith("E")) {
+                    String desc = descStr(input);
+                    String time = timeStr(input);
+                    LocalDate date = dateStr(input);
+                    if (String.valueOf(input.charAt(1)) == "Y") {
+                        tasks.add(new Event(true, desc, index, date, time));
+                    } else {
+                        tasks.add(new Event(false, desc, index, date, time));
+                    }
+                        index++;
+                        continue;
+                    }
+                }
+
+                sc.close();
+                return new TaskList(tasks, new Ui());
```

Comment:
Perhaps the indentation should be 4 spaces instead? I noticed the same indentation issue in some other places too.

Reply:
Noted, thank you very much!